client-v2/src/components/FrontsCAPIInterface/SearchInput.js

```diff
@@ -42,47 +89,140 @@ class FrontsCAPISearchInput extends React.Component<
     });
   };
 
-  handleTagInput = (item: any) => {
+  handleTagSearchInput = (
+    { currentTarget }: SyntheticEvent<HTMLInputElement>,
+    type: string
+  ) => {
+    const termInState = `${type}SearchTerm`;
     this.setState({
-      tag: item ? item.id : null
+      [termInState]: currentTarget.value
     });
   };
 
+  handleTagInput = (item: any, type: string) => {
+    const searchTerm = `${type}SearchTerm`;
+    const newTags = this.state[type];
+    if (item && newTags.indexOf(item.id) === -1) {
+      newTags.push(item.id);
+    }
+    this.setState({
+      [type]: newTags,
+      [searchTerm]: ''
+    });
+  };
+
+  handleDisplaySearchFilters = () => {
+    this.props.updateDisplaySearchFilters(!this.props.displaySearchFilters);
+  };
+
+  renderSelectedTags = (selectedTags: Array<string>) =>
+    selectedTags.map(searchTerm => (
+      <TagItem key={searchTerm}>
+        <span>{searchTerm}</span>
+        <SmallRoundButton
+          onClick={() => this.clearIndividualSearchTerm(searchTerm)}
+          title="Clear search"
+        >
+          <ClearButtonIcon
+            src={moreImage}
+            onClick={() => this.clearIndividualSearchTerm(searchTerm)}
+            alt=""
+            height="22px"
+            width="22px"
+          />
+        </SmallRoundButton>
+      </TagItem>
+    ));
+
   render() {
     const {
       children,
+      displaySearchFilters,
       additionalFixedContent: AdditionalFixedContent
     } = this.props;
-    const { tag, q } = this.state;
 
-    return (
-      <ScrollContainer
-        fixed={
-          <React.Fragment>
-            <InputContainer>
-              <TextInput
-                placeholder="Search"
-                value={this.state.q || ''}
-                onChange={this.handleSearchInput}
-                onClear={this.clearInput}
-              />
-            </InputContainer>
-            {AdditionalFixedContent && <AdditionalFixedContent />}
-          </React.Fragment>
-        }
-      >
-        <SearchQuery
-          params={{
-            tag,
-            q,
-            'show-elements': 'image',
-            'show-fields': 'internalPageCode,trailText'
-          }}
-          poll={30000}
+    const {
+      tags,
+      sections,
+      q,
+      tagsSearchTerm,
+      sectionsSearchTerm
+    } = this.state;
+
+    const searchTermsExist = tags.length !== 0 || sections.length !== 0 || !!q;
+    const tagQuery = tags ? tags.join(',') : '';
+    const sectionQuery = sections ? sections.join(',') : '';
+
+    if (!displaySearchFilters) {
```

Comment:
Is there any way we could merge these two cases a bit? If the `displaySearchFilters` check is done a few times at a more fine grained level it might mean we don't have to remember to change the props on things like `TextInput` in two places if they need changing in future?

Reply:
Yes! The second `displaySearchFilters` check was redundant.

Reply:
I missed that you fixed this because I'd commented on a redundant line! Sorrrry.